fdk/headers.py

```diff
@@ -21,9 +21,14 @@ def decap_headers(hdsr):
         for k, v in hdsr.items():
             k = k.lower()
             if k.startswith(constants.FN_HTTP_PREFIX):
-                ctx_headers[k.lstrip(constants.FN_HTTP_PREFIX)] = v
+                ctx_headers[k[len(constants.FN_HTTP_PREFIX):]] = v
             else:
-                ctx_headers[k] = v
+                # http headers override functions headers in context
+                # this is not ideal but it's the more correct view from the
+                # consumer perspective than random choice and for things
+                # like host headers
+                if k not in ctx_headers:
+                    ctx_headers[k] = v
     return ctx_headers
```

Comment:
Is it possible to add a tiny test for the decap_headers which test the headers override? It will guard this code for future changes, what do you think?

Reply:
I've added proper tests here, good call (also found a bug, thanks!)